### PR TITLE
Add LogAggregationContext

### PR DESCRIPTION
**Pull request by joaopamaral**
This change allows skip files from YARN log aggregation.

Manually tested

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -49,7 +49,7 @@ def _fix_imports(self, path):
 
     def run(self):
         from grpc_tools import protoc
-        include = pkg_resources.resource_filename('grpc_tools', '_proto')
+        include = resource_filename('grpc_tools', '_proto')
```
Comment (pbezglasny) [inline setup.py]: Not sure changes(replace import statement) in this file are really needed.

Comment (joaopamaral) [reply setup.py]: No need, probably my IDE changed it. I've reverted it

File: setup.py [hunk 2]
```diff
@@ -166,17 +166,13 @@ def run(self):
...
-    setup_requires = []
-
-
 install_requires = ['grpcio>=1.11.0,!=1.45.0',
                     'protobuf>=3.5.0',
                     'pyyaml',
                     'cryptography']
 
+setup_requires = ['grpcio-tools!=1.45.0']
+
```
Comment (joaopamaral) [inline setup.py:175]: this dynamic setup_requires was breaking the installation using pipenv

**Review by pbezglasny — APPROVED** [review]
LGTM

**Pull request merged**